apps/src/templates/sectionsRefresh/SectionsSetUpContainer.jsx

```diff
@@ -104,114 +109,118 @@ export default function SectionsSetUpContainer({sectionToBeEdited}) {
 
   const isNewSection = !sectionToBeEdited;
 
-  const caretStyle = style.caret;
   const caret = advancedSettingsOpen ? 'caret-down' : 'caret-right';
 
-  const toggleAdvancedSettingsOpen = () => {
-    setAdvancedSettingsOpen(!advancedSettingsOpen);
-  };
+  const toggleAdvancedSettingsOpen = useCallback(
```

Comment:
I'm sure I'm missing something but why is `useCallback` needed here?

Reply:
Good question, thank you!
Without  `useCallback` this function will be created every time when component is rendered which causes child elements being rerendered everytime the parent component is rendered. In this case it won't be drastic change, though avoiding it will still improve overall perfomance.

With `useCallback` this function will be memoized and will only be changed when one of the hooks dependenies will change. Meaning there won't be unnescecery rerenders of child component.

Here're couple articles about this hook if anyone's interested:
[https://react.dev/reference/react/useCallback](https://react.dev/reference/react/useCallback)
[w3schools.com/react/react_usecallback.asp](w3schools.com/react/react_usecallback.asp)

Reply:
ahhh because it's a prop to the child component! that makes a lot of sense thank you!